daemon.go

```diff
@@ -158,9 +158,9 @@ func (d *daemon) runCidCheck(ctx context.Context, cidKey cid.Cid, ipniURL string
 	queryCtx, cancelQuery := context.WithCancel(ctx)
 	defer cancelQuery()
 
-	// Find providers with DHT and IPNI concurrently.
-	provsCh := d.dht.FindProvidersAsync(queryCtx, cidKey, maxProvidersCount)
-	ipniProvsCh := routerClient.FindProvidersAsync(queryCtx, cidKey, maxProvidersCount)
+	// Find providers with DHT and IPNI concurrently (each half of the max providers count)
+	dhtProvsCh := d.dht.FindProvidersAsync(queryCtx, cidKey, maxProvidersCount>>1)
+	ipniProvsCh := routerClient.FindProvidersAsync(queryCtx, cidKey, maxProvidersCount>>1)
```

Comment:
I think if `maxProvidersCount` is not 0, then we should make sure it is at least 2 so that setting `maxProvidersCount = 1` continues to work.

Reply:
This is pretty unlikely since it's a constant. But I'll pushed a fix to handle this case.